src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -11339,20 +11546,745 @@ await using (new S2())
     }
 }
 ";
-            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics, options: TestOptions.ReleaseExe);
+            var comp1 = CreateCompilation(src1, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics, options: TestOptions.ReleaseExe);
 
-            comp.VerifyDiagnostics(
-                // PROTOTYPE(RefStructInterfaces): follow up on recent changes due to https://github.com/dotnet/csharplang/blob/main/proposals/ref-unsafe-in-iterators-async.md
-                // (18,22): error CS9104: A using statement resource of type 'S2' cannot be used in async methods or async lambda expressions.
+            var verifier = CompileAndVerify(
+                comp1, expectedOutput: ExecutionConditionUtil.IsMonoOrCoreClr ? @"123D" : null,
+                verify: ExecutionConditionUtil.IsMonoOrCoreClr ?
+                    Verification.Passes :
+                    Verification.Skipped).VerifyDiagnostics();
+
+            verifier.VerifyIL("C.<Main>d__0.System.Runtime.CompilerServices.IAsyncStateMachine.MoveNext",
+@"
+{
+  // Code size      235 (0xeb)
+  .maxstack  3
+  .locals init (int V_0,
+                S2 V_1,
+                object V_2,
+                System.Runtime.CompilerServices.ValueTaskAwaiter V_3,
+                System.Threading.Tasks.ValueTask V_4,
+                System.Exception V_5)
+  IL_0000:  ldarg.0
+  IL_0001:  ldfld      ""int C.<Main>d__0.<>1__state""
+  IL_0006:  stloc.0
+  .try
+  {
+    IL_0007:  ldloc.0
+    IL_0008:  brfalse.s  IL_0073
+    IL_000a:  ldloca.s   V_1
+    IL_000c:  initobj    ""S2""
+    IL_0012:  ldarg.0
+    IL_0013:  ldnull
+    IL_0014:  stfld      ""object C.<Main>d__0.<>7__wrap1""
+    IL_0019:  ldarg.0
+    IL_001a:  ldc.i4.0
+    IL_001b:  stfld      ""int C.<Main>d__0.<>7__wrap2""
+    .try
+    {
+      IL_0020:  ldc.i4.s   123
+      IL_0022:  call       ""void System.Console.Write(int)""
+      IL_0027:  leave.s    IL_0033
+    }
+    catch object
+    {
+      IL_0029:  stloc.2
+      IL_002a:  ldarg.0
+      IL_002b:  ldloc.2
+      IL_002c:  stfld      ""object C.<Main>d__0.<>7__wrap1""
+      IL_0031:  leave.s    IL_0033
+    }
+    IL_0033:  ldloca.s   V_1
+    IL_0035:  constrained. ""S2""
```

Comment:
Interesting. I didn't realize constrained calls worked in non generics.

Reply:
> Interesting. I didn't realize constrained calls worked in non generics. 

That is the way to call interface methods on structs without boxing